addons/mail/static/src/web/thread_service_patch.js

```diff
@@ -149,8 +149,11 @@ patch(ThreadService.prototype, "mail/web", {
     async insertSuggestedRecipients(thread, dataList) {
         const recipients = [];
         for (const data of dataList) {
-            const [partner_id, emailInfo, lang, reason] = data;
-            const [name, email] = emailInfo && parseEmail(emailInfo);
+            const [partner_id, emailInfo, lang, reason, customerInfo] = data;
+            let [name, email] = emailInfo && parseEmail(emailInfo);
+            if (name === email && customerInfo && customerInfo.name) {
```

Comment:
From 16.3 onwards you can simplify this line of code by using [optional chaining](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining)
```suggestion
            if (name === email && customerInfo?.name) {
```